### PR TITLE
Move pool into sat

### PR DESCRIPTION
**Pull request by cournape**
This migrates the Pool from `enstaller.new_solver` into `simplesat` (should be merged after `maintenance/0.1.x` branch is cut)

### CODE REVIEW

**Comment by cournape** [comment]
@johntyree first step to remove dependency on enstaller

File: simplesat/test_data/__init__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import os.path
```
Comment (johntyree) [inline simplesat/test_data/__init__.py]: Why not use `package_data` and `pkgutils` for this?

**Comment by johntyree** [comment]
Should remember to port https://github.com/enthought/enstaller_private/pull/638

File: simplesat/test_data/__init__.py (hunk 1 above)
Comment (cournape) [inline simplesat/test_data/__init__.py]: I don't like those packages, they incur too much complexity IMO.

Comment (johntyree) [inline simplesat/test_data/__init__.py]: You've already added these to the package_data in `setup.py`, though, so it should already "just work."

We're already using that method for the scenario yaml files *and* it claims to be robust against zipped eggs. I also found it pretty easy to use while setting up airspeed-velocity for benchmarks.

Having a second way to compute (sometimes correct) paths won't interfere with that, but using this requires that eggs be unzipped, doesn't it?

**Comment by johntyree** [comment]
I think this is alright as-is. I'll just complain about not using `pkgutil.get_data` in future PRs. :wink:

**Comment by johntyree** [comment]
Oh hmm. Meant to do this after making the maintenance branch...

I suppose I'll just have to make that branch from the parent of this.

**Pull request merged**